src/LoggerManager.ts

```diff
@@ -0,0 +1,95 @@
+import { Client, Intents, ClientEvents } from 'discord.js';
+
+import { EventParser } from './EventParser';
+import { IEventParser, IModule, ModuleMethod } from '@types';
+
+export class LoggerManager<Event extends keyof ClientEvents> {
+	private readonly _bindedModuleMethodsPerModuleMethods: Map<
+		ModuleMethod,
+		Map<Partial<IModule>, ModuleMethod>
+	>;
+	private readonly _modules: Array<Partial<IModule>>;
+	private readonly _client: Client;
+	private readonly _eventParser: IEventParser;
+
+	constructor(client: Client, modules: Array<Partial<IModule>> = []) {
+		if (!client) throw new Error('You must provide a client!');
+
+		this._client = client;
+		this._modules = modules;
+		this._bindedModuleMethodsPerModuleMethods = new Map<
+			ModuleMethod,
+			Map<Partial<IModule>, ModuleMethod>
+		>();
+		this._eventParser = new EventParser();
+	}
+
+	public listenWithAllModulesTo(event: Event) {
+		if (this._client.eventNames().includes(event)) return;
```

Comment:
Same comment: might need to be removed because we should still be able to listen to events from the client without altering the behaviour of this method.